lms/djangoapps/mobile_api/utils.py

```diff
@@ -30,12 +34,28 @@ def _wrapper(self, request, *args, **kwargs):
             """
             course_id = CourseKey.from_string(kwargs.pop('course_id'))
             with modulestore().bulk_operations(course_id):
-                course = get_course_with_access(
-                    request.user,
-                    'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
-                    course_id,
-                    depth=depth
-                )
+                try:
+                    course = get_course_with_access(
+                        request.user,
+                        'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
+                        course_id,
+                        depth=depth
+                    )
+                except Http404 as error:
+                    # any_unfulfilled_milestones called a second time since get_course_with_access returns a bool
+                    if any_unfulfilled_milestones(course_id, request.user.id):
+                        message = {
+                            "developer_message": "Cannot access content with unfulfilled pre-requisites or unpassed entrance exam."  # pylint: disable=line-too-long
+                        }
+                        return response.Response(
+                            data=message,
+                            status=status.HTTP_204_NO_CONTENT
+                        )
+                    else:
+                        return response.Response(
+                            data=error.message,
+                            status=status.HTTP_404_NOT_FOUND
+                        )
```

Comment:
To re-raise an exception in python, just use `raise`.  So you can replace lines 55-58 with `raise`.